src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/features/OcrDataSerializationJourneyTest.java

```diff
@@ -0,0 +1,73 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.features;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.io.IOUtils;
+import org.assertj.core.api.AssertionsForInterfaceTypes;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.context.junit4.SpringRunner;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.Envelope;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.EnvelopeRepository;
+import uk.gov.hmcts.reform.bulkscanprocessor.helper.EnvelopeCreator;
+import uk.gov.hmcts.reform.bulkscanprocessor.model.blob.InputEnvelope;
+import uk.gov.hmcts.reform.bulkscanprocessor.model.mapper.EnvelopeMapper;
+import uk.gov.hmcts.reform.bulkscanprocessor.model.out.msg.EnvelopeMsg;
+
+import java.io.InputStream;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.UUID;
+
+@RunWith(SpringRunner.class)
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+@DataJpaTest
+public class OcrDataSerializationJourneyTest {
+
+    @Autowired
+    private EnvelopeRepository repository;
+
+    @Test
+    public void should_deserialize_all_ocr_fields_in_insertion_order() throws Exception {
```

Comment:
This test doesn't test deserialisation. It only test if the ocrData map in ScannableItem converts to the right JSON in the message. The rest of the journey is irrelevant.

Reply:
it is a journey of ocr data. we don't really care bits in between what matters is input and output stays the same. which got somehow missed out 🤔